moped-editor/src/views/projects/projectView/ProjectTeamTable.js

```diff
@@ -153,44 +136,95 @@ const ProjectTeamTable = ({
           value={props.value}
           onChange={e => props.onChange(e.target.value)}
         />
       ),
     },
   ];
 
+  /**
+   * Data handlers for editable actions based on isNewProject boolean <MaterialTable>
+   */
+  const isNewProjectActions = {
+    true: {
+      add: newData => {
+        const activePersonnel = { ...newData, status_id: 1 };
+
+        setPersonnelState([...personnelState, activePersonnel]);
+      },
+      update: (newData, oldData) => {
+        const dataUpdate = [...personnelState];
+        const index = oldData.tableData.id;
+        dataUpdate[index] = newData;
+        setPersonnelState([...dataUpdate]);
+      },
+      delete: oldData => {
+        const dataDelete = [...personnelState];
+        const index = oldData.tableData.id;
+        dataDelete.splice(index, 1);
+        setPersonnelState([...dataDelete]);
+      },
+    },
+    false: {
+      add: newData => {
+        const personnelData = {
+          ...newData,
+          project_id: projectId,
+          status_id: 1,
+        };
+
+        addProjectPersonnel({
+          variables: {
+            objects: [personnelData],
+          },
+        });
+      },
+      update: (newData, oldData) => {
+        const updatedPersonnelData = {
+          ...oldData,
+          ...newData,
+        };
+
+        const cleanedPersonnelData = filterObjectByKeys(updatedPersonnelData, [
+          "__typename",
+          "tableData",
+        ]);
+
+        updateProjectPersonnel({
+          variables: cleanedPersonnelData,
+        });
+      },
+      delete: oldData => {
+        const updatedPersonnelData = { ...oldData, status_id: 0 };
+
+        const cleanedPersonnelData = filterObjectByKeys(updatedPersonnelData, [
+          "__typename",
+          "tableData",
+        ]);
+
+        updateProjectPersonnel({
+          variables: cleanedPersonnelData,
+        });
+      },
+    },
+  };
+
   return (
     <MaterialTable
       columns={columns}
-      data={personnel}
+      data={isNewProject ? personnelState : personnel}
       title="Project Team"
       options={{
         search: false,
+        rowStyle: { fontFamily: typography.fontFamily },
       }}
       icons={{ Delete: ClearIcon }}
       editable={{
         onRowAdd: newData =>
           new Promise((resolve, reject) => {
             setTimeout(() => {
-              if (isNewProject) {
-                // Add personnel to state
-                console.log("Add to new project");
-              } else {
-                // Insert personnel and associate with project
-                const personnelData = {
-                  ...newData,
-                  project_id: projectId,
-                  status_id: 1,
-                };
-
-                addProjectPersonnel({
-                  variables: {
-                    objects: [personnelData],
-                  },
-                });
-              }
+              isNewProjectActions[isNewProject].add(newData);
```

Comment:
This is classy! I like how you used true and false as keys, never thought you could do that :) 👍

Reply:
Same here, it was a nice surprise that it worked out this way. 😄